cmake/MOAB_PullAndMake.cmake

```diff
@@ -0,0 +1,86 @@
+# This Macro sets up the download and build of MOAB using ExternalProject
+# few tweak are done in src/dagmc/CMakeLists.txt and src/PyNE/CMakeists.txt 
+# to make sure that MOAB is built before DAGMC.
+MACRO (moab_pull_make MOAB_Version)
+  MESSAGE(STATUS "MOAB will be downloaded and built")
+  
+  # Ensure STATIC LIBS Build
+  IF(DAGMC_BUILD_STATIC_LIBS)
+    MESSAGE(FATAL_ERROR "PULL_INSTALL_MOAB is ONLY compatible with shared libraries.")
+  endIF()
+    
+  # Ensure we have the right variables to build MOAB
+  moab_autobuild_check_deps()
+   
+    
+  INCLUDE(ExternalProject)
+  MESSAGE("HDF5_ROOT: ${HDF5_ROOT}")
+  SET(MOAB_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/moab")
+  SET(MOAB_ROOT "${CMAKE_BINARY_DIR}/moab")
+  SET(MOAB_INCLUDE_DIRS "${MOAB_INSTALL_PREFIX}/include")
+  SET(MOAB_LIBRARY_DIRS "${MOAB_INSTALL_PREFIX}/lib")
+  MESSAGE("MOAB_LIBRARY_DIRS: ${MOAB_LIBRARY_DIRS}")
+  MESSAGE("CMAKE_SHARED_LIBRARY_SUFFIX: ${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  SET(MOAB_LIBRARIES_SHARED "")
+  ExternalProject_Add(MOAB_ep
+    PREFIX ${MOAB_ROOT}   
+    GIT_REPOSITORY https://bitbucket.org/fathomteam/moab.git
+    GIT_TAG ${moab_version}
+    CMAKE_ARGS
+      -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+      -DCMAKE_BUILD_TYPE=Release
+      -DENABLE_HDF5:BOOL=ON
+      -DHDF5_ROOT:PATH=${HDF5_ROOT}
+      -DCMAKE_INSTALL_RPATH=${HDF5_ROOT}/lib:${MOAB_INSTALL_PREFIX}/lib
+      -DENABLE_BLASLAPACK:BOOL=OFF
+      -DBUILD_SHARED_LIBS:BOOL=ON
```

Comment:
There's an option to build both static and shared libraries with DAGMC. To build with this option enabled tn the past, I've had to build MOAB twice -- once with this value set to `ON` and again with it set to `OFF`.

I don't think this is a common use case these days though and I'd probably suggest that if `PULL_INSTALL_MOAB` is on that we check if static libraries are requested and error with a notification that these settings (`PULL_INSTALL_MOAB=ON` and `BUILD_STATIC_LIBS=ON`) aren't compatible.

Reply:
there should already be a safeguard for this case l9-10 of this file.

Reply:
Ah, right you are. Thanks!